GitUI/ScriptsEngine/IScriptsRunner.cs

```diff
@@ -5,10 +5,8 @@ namespace GitUI.ScriptsEngine
 {
     public interface IScriptsRunner
     {
-        bool RunEventScripts<THostForm>(ScriptEvent scriptEvent, THostForm form)
-            where THostForm : IGitModuleForm, IWin32Window;
+        bool RunEventScripts(ScriptEvent scriptEvent, IScriptHostControl scriptHostControl);
```

Comment:
Reverting this will save lots of `new DefaultScriptHostControl(this, UICommands)` and reduce the change-set.

Reply:
It would also mean `IScriptHostControl` can't simply have `IWin32Window` and `IGitUICommands`, and, instead, require it to have the DI-unfriendly `GitModuleForm` (the closest common child of `IGitModuleForm` and `IWin32Window`). From the get-go, there was no reason to pass both as one parameter, since `IGitModuleForm` is only used for its `UICommands` property and nothing else.

An alternative would be to pass it as three parameters, like in `RunScriptInternal`, but you actually suggested compressing it there.

Reply:
What is missing for `RunEventScripts<THostForm>(ScriptEvent scriptEvent, THostForm form)` in order to create a `DefaultScriptHostControl` itself? There should be no need for a closest common child.

Reply:
Ah, I didn't notice it's `RunEventScripts` (which skips the need for a dedicated `IScriptHostControl`) and not `RunScripts` (which needs to drop the `THostForm` due to the call in `UserScriptContextMenuExtensions.AddUserScripts`).

Accepted